app/src/main/java/com/google/droidjump/GameFailureFragment.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.droidjump;
+
+import android.os.Bundle;
+import android.support.design.widget.FloatingActionButton;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageButton;
+
+import androidx.navigation.Navigation;
+
+import java.util.Objects;
+
+
+public class GameFailureFragment extends Fragment {
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.game_failure_screen, container, false);
+        FloatingActionButton retryButton = rootView.findViewById(R.id.retry_button);
+        retryButton.setOnClickListener(view -> {
+            Navigation.findNavController(view).navigate(R.id.action_game_failure_screen_to_game_screen);
+        });
+        ImageButton menuButton = rootView.findViewById(R.id.success_menu_button);
+        menuButton.setOnClickListener(view -> {
+            Objects.requireNonNull(getActivity()).onBackPressed();
```

Comment:
hmm, I am  bit confused, why do we use onBackPressed to go back to the menu instead of going directly to the menu screen?

Reply:
In navigation.xml I created connections between all the screens. When you press back button it redirects to StartFragment(Screen with the main menu).

Reply:
@redls do you think it will be better to create a new action for this?

Reply:
Or just add the comment for better understanding?

Reply:
if possible, I would create a new action. Otherwise, a comment will definitely help

Reply:
Thanks)